addOns/spider/src/test/java/org/zaproxy/addon/spider/parser/SpiderHtmlParserUnitTest.java

```diff
@@ -685,8 +777,8 @@ void shouldNotFindUrlsInCommentsWithElementsIfNotEnabledToParseComments() {
         boolean completelyParsed = parser.parseResource(ctx);
         // Then
         assertThat(completelyParsed, is(equalTo(false)));
-        assertThat(listener.getNumberOfUrlsFound(), is(equalTo(0)));
-        assertThat(listener.getUrlsFound(), is(empty()));
+        assertThat(listener.getNumberOfUrlsFound(), is(equalTo(1)));
+        assertThat(listener.getUrlsFound(), contains("http://example.com/sample/"));
```

Comment:
With this PR we are parsing the url found in base tag's href attribute, which we were not doing before. And since the html file has a base tag with a valid href attribute and url, it finds it. If we strictly want this particular function (`shouldNotFindUrlsInCommentsWithElementsIfNotEnabledToParseComments`) to not report any links, we can remove the base tag from the html file since it is not doing anything anyway.

Reply:
I mean we were parsing base tag  before but only to use that properly parse the other found links on the page.

Reply:
Well I see two ways to go:
1. Make it return nothing since that's what's suggested.
2. Make it really clear that was was found was in the base tag. `"http://example.com/sample/"` :arrow_right: `"http://example.com/from/base/tag/"`